app/controllers/payments_controller.rb

```diff
@@ -0,0 +1,61 @@
+class PaymentsController < ApplicationController
+
+  before_action :authenticate_user!
+  load_and_authorize_resource
+  before_action :find_invoice
+  before_action :replace_date_params , only: [:create,:update]
+
+  def index
+    @payments = @invoice.payments
+  end
+
+  def new
+    @payment = Payment.new
+  end
+
+  def edit
+    @payment = Payment.find(params[:id])
+  end
+
+  def create
+    @payment = @invoice.payments.build(payment_params)
+    if @payment.save
+      begin
+        ClientMailer.send_email_payment(@invoice, current_user).deliver if params[:email].present?
+        flash[:success] = 'Payment saved successfully.'
```

Comment:
This message should be outside `being .. rescue`. So use should be aware that the payment was successful but unable to sent email.